src/main/groovy/com/brightsparklabs/gradle/baseline/BaselinePlugin.groovy

```diff
@@ -351,45 +358,139 @@ public class BaselinePlugin implements Plugin<Project> {
+                    " `${bucketName}`"
             project.logger.error(error)
             throw new IllegalStateException(error)
         }
         if (filesToUploadIsEmpty) {
-            def error = "`bslGradle.deploy.s3.filesToUpload` cannot be null or empty. Value was: ${filesToUpload}"
+            def error = "`bslGradle.deploy.s3.filesToUpload` cannot be null or empty. Value was:" +
+                    " ${filesToUpload}"
             project.logger.error(error)
             throw new IllegalStateException(error)
         }
 
+        // Get the absolute paths of the files to upload. We search for files in the `build`
+        // directory, and pattern match against their absolute paths.
+        final Set<String> filesToUploadPaths = []
+        for (String fileRegex in filesToUpload) {
+            project.buildDir.eachFileRecurse(FileType.FILES) { File file ->
+                if (file.absolutePath ==~ fileRegex) {
+                    filesToUploadPaths.add(file.absolutePath)
+                }
+            }
+        }
+        for (String filePathString in filesToUploadPaths) {
+            project.logger.lifecycle("Found file to upload: `${filePathString}`")
+        }
+
+
         project.task("bslDeployToS3") {
             group = "brightSPARK Labs - Baseline"
-            description = "Upload files to an S3 bucket. Configure via the `bslBaseline` configuration block."
+            description = "Upload files to an S3 bucket. Configure via the `bslBaseline`" +
+                    " configuration block."
 
             doLast {
                 final S3ClientBuilder s3Builder = S3Client.builder()
+
                 // By default, the AWS SDK will attempt to pull the region from the system.
                 // If configured, we allow for an optional override.
                 if (!Strings.isNullOrEmpty(region)) {
                     s3Builder.region(Region.of(region))
                 }
-                final S3Client s3 = s3Builder.build()
+
+                // By default, the AWS SDK will use the "default" profile from the system.
+                // If configured, we allow for an optional override.
+                if (!Strings.isNullOrEmpty(profile)) {
+                    final ClientOverrideConfiguration.Builder s3OverrideConfigBuilder =
+                            ClientOverrideConfiguration.builder()
+                    s3OverrideConfigBuilder.defaultProfileName(profile)
+                    final s3OverrideConfig = s3OverrideConfigBuilder.build()
+                    s3Builder.overrideConfiguration(s3OverrideConfig)
+                }
+
+                // By default, the AWS SDK will use the default S3 endpoint for the given region.
+                // If configured, we allow for an optional override.
+                if (!Strings.isNullOrEmpty(endpointOverride)) {
+                    URI endpointOverrideURI
+                    try {
+                        endpointOverrideURI = new URI(endpointOverride)
```

Comment:
You might be able to clean this up with try with resources:
- https://www.baeldung.com/java-try-with-resources

```
try (URI  endpointOverrideURI = new URI(endpointOverride) {
  s3Builder.endpointOverride(endpointOverrideURI)
} catch (...)
```

Not entirely certain as URI doesn't need resources, but worth seeing for the cleaner code.

Reply:
Unfortunately that doesn't work since `URI` doesn't implement `AutoCloseable`.